src/core/elements/importElements.js

```diff
@@ -68,31 +50,26 @@ const importElements = curry(async (elements, account, options) => {
   }
 });
 
-module.exports = (account, options) => {
+module.exports = async (account, options) => {
   try {
-    return cond([
-      [
-        pipe(prop('file'), isNil, not),
-        pipeP(
-          useWith(readFile, [prop('file')]),
-          cond([
-            [
-              pipe(type, equals('Object')) && pipe(prop('elements'), isNil, not),
-              pipe(prop('elements'), importElements(__, account, options)),
-            ],
-            [pipe(type, equals('Array')), importElements(__, account, options)],
-          ]),
-        ),
-      ],
-      [
-        pipe(prop('dir'), isNil, not),
-        pipeP(useWith(buildElementsFromDir, [prop('dir')]), importElements(__, account, options)),
-      ],
-    ])(options);
+    let elementsDataToImport;
+    if (pipe(prop('file'), isNil, not)(options)) {
+      elementsDataToImport = await readFile(options.file);
+    } else if (pipe(prop('dir'), isNil, not)(options)) {
+      elementsDataToImport = await readElementsFromDir(options.dir);
+    }
+    if (isNilOrEmpty(elementsDataToImport)) {
+      logError(`No elements found for import operation`);
+      return;
+    }
+    if (pipe(type, equals('Object')) && pipe(prop('elements'), isNil, not)(elementsDataToImport)) {
+      /* istanbul ignore next */
+      await importElements(pipe(prop('elements'))(elementsDataToImport), account, options);
```

Comment:
**nitpick:** We can do away without `pipe` here and in other operation files

Reply:
We can do that but I just wanted to keep this code as it's more readable for me.